### PR TITLE
OCPBUGS-36525: Bump k8s to 1.30.2

### PR DESCRIPTION
**Pull request by ingvagabund**
Including:
- bump of github.com/moby/buildkit and github.com/containerd/containerd
- update pkg/helpers/newapp/docker/dockerfile code to address changed from github.com/moby/buildkit

### CODE REVIEW

File: images/cli/Dockerfile.rhel [hunk 1]
```diff
@@ -8,5 +8,5 @@ COPY --from=builder /go/src/github.com/openshift/oc/oc /usr/bin/
 RUN for i in kubectl openshift-deploy openshift-docker-build openshift-sti-build openshift-git-clone openshift-manage-dockerfile openshift-extract-image-content openshift-recycle; do ln -sf /usr/bin/oc /usr/bin/$i; done
 LABEL io.k8s.display-name="OpenShift Client" \
       io.k8s.description="OpenShift is a platform for developing, building, and deploying containerized applications." \
-      io.openshift.build.versions="kubectl=1.30.1" \
+      io.openshift.build.versions="kubectl=1.30.2" \
```
Comment (ardaguclu) [inline images/cli/Dockerfile.rhel:11]: We need to update tools image as well https://github.com/openshift/oc/blob/5e45e289cc56aa7a29533c8438e1fe72ee6de87c/images/tools/Dockerfile#L57

**Review by ardaguclu — COMMENTED** [review]
That looks good to me. Thank you.

File: images/cli/Dockerfile.rhel (hunk 1 above)
Comment (ingvagabund) [reply images/cli/Dockerfile.rhel:11]: Updated. Thx.

**Comment by ardaguclu** [comment]
@ingvagabund can we also get this https://github.com/openshift/oc/pull/1810/files#diff-574428bc9aabd6570030124166cdb957bfa305a098379edf053a88a1d0e772f2R19-R23 in here. It is useful.

**Comment by ingvagabund** [comment]
> can we also get this https://github.com/openshift/oc/pull/1810/files#diff-574428bc9aabd6570030124166cdb957bfa305a098379edf053a88a1d0e772f2R19-R23 in here. It is useful.

https://github.com/openshift/oc/pull/1813

**Pull request merged**